controllers/release/release_adapter_test.go

```diff
@@ -226,6 +275,296 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		Expect(k8sClient.Delete(ctx, pipelineRun)).Should(Succeed())
 	})
 
+	Context("When calling EnsureSnapshotEnvironmentBindingIsCreated", func() {
+		It("skips the operation if the release has not succeeded yet", func() {
+			result, err := adapter.EnsureSnapshotEnvironmentBindingIsCreated()
+			Expect(result.RequeueRequest).To(BeFalse())
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("skips the operation if the release has already being deployed", func() {
+			release.MarkRunning()
+			release.MarkSucceeded()
+			release.Status.SnapshotEnvironmentBinding = "foo"
+			result, err := adapter.EnsureSnapshotEnvironmentBindingIsCreated()
+			Expect(result.RequeueRequest).To(BeFalse())
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("fails when the required resources are not present", func() {
+			release.MarkRunning()
+			release.MarkSucceeded()
+
+			// Delete the application which is one of the required resources
+			Expect(k8sClient.Delete(ctx, application)).Should(Succeed())
```

Comment:
I would add an Eventually block just to garantee. Noticed that lacking it might add flakiness in the tests.

Reply:
I agree. I added it in some other places but I forgot about this one. Good catch.